pkg/network/tcp_peer.go

```diff
@@ -128,9 +189,14 @@ func (p *TCPPeer) StartProtocol() {
 	for {
 		select {
 		case <-p.done:
-			break
+			return
```

Comment:
I think this change could be done in a previous commit, where this line was first written.

Reply:
Absolutely true, fixed that.